pkg/controller/atlasproject/private_endpoint.go

```diff
@@ -585,3 +598,31 @@ func hasSkippedPrivateEndpointConfiguration(atlasProject *akov2.AtlasProject) (b
 
 	return false, nil
 }
+
+func mapLastAppliedPrivateEndpoint(atlasProject *akov2.AtlasProject) (map[string]akov2.PrivateEndpoint, error) {
+	lastApplied, err := lastSpecFrom(atlasProject, customresource.AnnotationLastAppliedConfiguration)
+	if err != nil {
+		return nil, err
+	}
+
+	if lastApplied == nil || len(lastApplied.PrivateEndpoints) == 0 {
+		return nil, nil
+	}
+
+	result := map[string]akov2.PrivateEndpoint{}
+	for _, pe := range lastApplied.PrivateEndpoints {
+		result[pe.Identifier().(string)] = pe
+	}
+
+	return result, nil
+}
+
+func hasManagedPrivateEndpoints(specPEs []akov2.PrivateEndpoint, atlasPEs []atlasPE, lastAppliedPEs map[string]akov2.PrivateEndpoint) bool {
+	for _, pe := range atlasPEs {
+		if _, ok := lastAppliedPEs[pe.Identifier().(string)]; ok {
+			return true
+		}
+	}
+
+	return len(set.DeprecatedDifference(specPEs, atlasPEs)) == 0
```

Comment:
why are we using a deprecated diff on new code?

Reply:
As mentioned in previous comments, while there's new addition to this part of the code, it still depends on deprecated types/interfaces unless fully refactor to use translation layer, which is not a goal.